Source/WTF/wtf/win/OSAllocatorWin.cpp

```diff
@@ -146,7 +146,7 @@ bool OSAllocator::protect(void* address, size_t bytes, bool readable, bool writa
         return VirtualAlloc(address, bytes, MEM_COMMIT, protection);
     }
     ASSERT(!readable && !writable);
-    return VirtualFree(address, bytes, MEM_DECOMMIT);
+    return VirtualAlloc(address, bytes, MEM_COMMIT, PAGE_NOACCESS);
```

Comment:
If we're going to switch to always calling VirtualAlloc just with different protection values (which seems closer to what the posix one does anyway), it might be good to format it more like that one as well rather than the early exit in the readable block. https://github.com/WebKit/WebKit/blob/main/Source/WTF/wtf/posix/OSAllocatorPOSIX.cpp#L285-L298